src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -115,7 +118,10 @@ public void setAccessModifiers(AccessModifier... accessModifiers) {
 
     @Override
     public int[] getRequiredTokens() {
-        return new int[] {TokenTypes.PARAMETER_DEF};
+        return new int[] {
+            TokenTypes.PARAMETER_DEF,
+            TokenTypes.LAMBDA,
```

Comment:
@romani Should it be possible to specify different naming conventions between parameters and lambdas?
If so, this shouldn't be a required field, but be acceptable and default.

Reply:
I agree with @rnveach . Method parameters usually have broader scope then lambda parameters. For example, I want to disallow single character names for method parameters and allow them for lambdas, because lambda block usually contains small number of lines, and it is easy to hold the context of it in mind.

Reply:
strange for some reason my comment disappeared ...

During review I was ok to make them as required, as parameter in lambda and method are kind of the same term, so I was ok. 
But you are right, look at http://checkstyle.sourceforge.net/config_naming.html we are very fine grained on what we validate. So user have bunch of options to allow special cases/names for `catch` or ...... .
Yes, lets make Lambda as non required but as default. So by default most user will not bother with extra details (the same for google style, no changes). But smb who want to make different rules for lambdas (probably some allowances for single character names) could be possible (even not recommended).

@kazachka , please make this change.
Please also add example to xdoc on how to make rules different for method parameters and lambda.

Reply:
@kazachka Nothing should be listed in required as we need to be able to turn on each one individually. Acceptable/default should return both token types.